src/tracing.rs

```diff
@@ -61,6 +61,15 @@ pub fn open_telemetry<S>(
 where
     S: Subscriber + for<'span> LookupSpan<'span>,
 {
+    let disabled = env::var("OTEL_DISABLED")
+        .ok()
+        .unwrap_or_default()
+        .contains("true");
+
+    if disabled {
+        return Ok(OpenTelemetryLayer::new(otlp_tracer(&[])?).with_filter(LevelFilter::OFF));
```

Comment:
1. Please log that it's disabled
2. The `otlp_tracer` function should never be called, you can use a noop tracer instead